File: Examples/test-suite/python/smart_pointer_member_runme.py
from smart_pointer_member import *


f = Foo()
f.y = 1

if f.y != 1:
    raise RuntimeError

b = Bar(f)
b.y = 2

if f.y != 2:
    print f.y
    print b.y
    raise RuntimeError

if b.x != f.x:
    raise RuntimeError

if b.z != f.z:
    raise RuntimeError

if Foo.z == Bar.z:
    raise RuntimeError

File: Examples/test-suite/lua/smart_pointer_member_runme.lua
local v=require("smart_pointer_member")

local f = v.Foo()
f.y = 1
assert(f.y == 1)

b = v.Bar(f)
b.y = 2
assert(f.y == 2)
assert(swig_equals(b.x, f.x))
assert(b.z == v.Foo.z)
assert(swig_equals(b:__deref__(), f))
